download-wgpu-native.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+import os
+import requests
+import sys
+import tempfile
+from zipfile import ZipFile
+
+
+RESOURCE_DIR = os.path.join("wgpu", "resources")
+VERSION_FILE = os.path.join(RESOURCE_DIR, "wgpu_native-version")
+
+
+def get_current_version():
+    with open(VERSION_FILE, mode="r") as fh:
+        return fh.read()
+
+
+def write_current_version(version):
+    with open(VERSION_FILE, mode="w") as fh:
+        return fh.write(version)
+
+
+def download_file(url, filename):
+    CHUNK_SIZE = 1024 * 128
+    resp = requests.get(url, stream=True)
+    with open(filename, mode="wb") as fh:
+        for chunk in resp.iter_content(chunk_size=CHUNK_SIZE):
+            fh.write(chunk)
+
+
+def extract_files(zip_filename, members, path, missing_ok=True):
+    z = ZipFile(zip_filename)
+    for member in members:
+        try:
+            z.extract(member, path=path)
+        except KeyError:
+            if not missing_ok:
+                raise
+
+
+def get_os_string():
+    if sys.platform.startswith("win"):
+        return "windows"
+    elif sys.platform.startswith("darwin"):
+        return "macos"
+    elif sys.platform.startswith("linux"):
+        return "linux"
+
+
+def main(version, debug, os_string, upstream):
+    debug = "debug" if debug else "release"
+    filename = f"wgpu-{debug}-{os_string}-{version}.zip"
+    url = f"https://github.com/{upstream}/releases/download/{version}/{filename}"
+    tmp = tempfile.gettempdir()
+    zip_filename = os.path.join(tmp, filename)
+    print(f"Downloading {url} to {zip_filename}")
+    download_file(url, zip_filename)
+    members = [
+        "wgpu.h",
+        "libwgpu_native.so",
+        "libwgpu_native.dylib",
+        "wgpu_native.dll",
+        "commit-sha",
+    ]
+    print(f"Extracting {members} to {RESOURCE_DIR}")
+    extract_files(zip_filename, members, RESOURCE_DIR)
+    current_version = get_current_version()
+    if version != current_version:
+        print(f"Version changed, updating {VERSION_FILE}")
+        write_current_version(version)
```

Comment:
So, if I got this right, the file `wgpu_native-version` is in source control, and with it we specify what version we want. And if we use this script to update the version, we also update it as a marker to show what lib is currently in the resource dir. Can we have a description like this in a comment somewhere?

Reply:
That's right. I would personally not edit it manually at all, and always use the script, but there is nothing stopping you from doing so. :) I'll add instructions to README.md.

Reply:
Right, you can edit it by running the script ;)